client/src/main/scala/org/http4s/netty/client/Http4sHandler.scala

```diff
@@ -81,13 +77,21 @@ private[netty] class Http4sHandler[F[_]](dispatcher: Dispatcher[F])(implicit F:
   private def safedispatch(
       request: HttpRequest,
       key: RequestKey,
-      callback: Either[Throwable, Resource[F, Response[F]]] => Unit): Runnable = () =>
-    void {
+      callback: Either[Throwable, Resource[F, Response[F]]] => Unit): Unit = void {
+    val ch = ctx.channel
+    if (ch.isActive) {
       promises.enqueue(callback)
-      logger.trace(s"Sending request to $key")
-      ctx.writeAndFlush(request).sync()
```

Comment:
Note that this line was very prone to deadlocks: if the whole message wasn't able to be flushed immediately the channels thread would end up blocking on the `.sync()` call, despite also being responsible for finishing the flushing of said message.